### ISSUE TITLE
seeing recursion with DTCollectionViewDelegate canFocus

### ISSUE THREAD

**Issue by skydivedan** [DenHeadless/DTTableViewManager#64]
Sorry, this isn't really an issue, but asking for advice, as to whether you've seen this and might know the cause. 

There is a situation (which I'm still trying to determine the cause) where `canFocus` gets called repeatedly, and recursively. Screenshot attached of stacktrace. I'm sure it's some dumb issue on my part, just not sure best way to proceed.

<img width="357" alt="screen shot 2018-10-24 at 12 20 12 pm" src="https://user-images.githubusercontent.com/217969/47451357-89650c80-d795-11e8-9633-efb29216bc6c.png">

**Comment by DenHeadless** [comment]
Hey!

I believe this might be a bug in DTCollectionViewManager. From documentation on UICollectionViewDelegate `collectionView(_:canFocusItemAt:)`:

```
You can use this method, or a cell’s canBecomeFocused method, to control which items in the collection view can receive focus. The focus engine calls the cell’s canBecomeFocused method first, the default implementation of which defers to the collection view and this delegate method.
```

Which is unfortunate, because this event implementation in DTCollectionViewManager is implemented like so:
https://github.com/DenHeadless/DTCollectionViewManager/blob/master/Source/DTCollectionViewDelegate.swift#L181

And so what I was trying to do is 
1. Call closure event, if it's registered
2. Call real UICollectionViewDelegate method, if it's implemented
3. Call cell's canBecomeFocused.
4. Return true if all other steps fail as a sensible default.

What seems to be happening is step 3 implicitly calls step 2 in UICollectionView implementation which leads to infinite recursion. 

I don't have experience of working with this API, but it seems like removing step 3 in DTCollectionViewDelegate implementation looks like the way to go. What do you think?

**Comment by skydivedan** [comment]
Yes, your assessment looks correct. Additionally, you'll have to take out the same clause in `DTTableViewDelegate.canFocusRowAt`. I'm going to run some tests here, and see what develops.

**Comment by skydivedan** [comment]
Quick note... sorry to say but I can't get back to this until Monday, unfortunately. But I dug around a bit more in the code and found a similar example of this `DTTableViewDelegate` in `tableView.shouldIndentWhileEditingRowAt`.

**Comment by DenHeadless** [comment]
@skydivedan No worries, take your time, it can wait till Monday =)

As for `tableView.shouldIndentWhileEditingRowAt` - it's different because UITableViewCell does not actually call delegate method if you ask it's property `shouldIndentWhileEditing`, and 4 steps logic I described earlier should work fine.

**Comment by DenHeadless** [comment]
This bug is fixed in [DTCollectionViewManager 6.4.1 release](https://github.com/DenHeadless/DTCollectionViewManager/releases/tag/6.4.1). Closing.
